extmod/modbluetooth.c

```diff
@@ -172,6 +172,37 @@ STATIC void bluetooth_uuid_print(const mp_print_t *print, mp_obj_t self_in, mp_p
     }
 }
 
+mp_int_t bluetooth_uuid_get_buffer(mp_obj_t self_in, mp_buffer_info_t *bufinfo, mp_uint_t flags) {
+    mp_obj_bluetooth_uuid_t *self = MP_OBJ_TO_PTR(self_in);
+
+    if (flags == MP_BUFFER_READ) {
+        uint8_t *buf = m_new(uint8_t, self->type);
+        if (self->type == MP_BLUETOOTH_UUID_TYPE_16) {
+            buf[0] = self->uuid._16 & 0xff;
+            buf[1] = (self->uuid._16 >> 8) & 0xff;
+        } else if (self->type == MP_BLUETOOTH_UUID_TYPE_32) {
+            buf[0] = self->uuid._32 & 0xff;
+            buf[1] = (self->uuid._32 >> 8) & 0xff;
+            buf[2] = (self->uuid._32 >> 16) & 0xff;
+            buf[3] = (self->uuid._32 >> 24) & 0xff;
+        } else if (self->type == MP_BLUETOOTH_UUID_TYPE_128) {
+            for (int i = 0; i < 16; ++i) {
+                memcpy(buf, self->uuid._128, 16);
+            }
+        }
+        bufinfo->buf = buf;
+        bufinfo->len = self->type;
+        bufinfo->typecode = 'B';
+        return 0;
+    } else {
+        // can't write to a string
+        bufinfo->buf = NULL;
+        bufinfo->len = 0;
+        bufinfo->typecode = -1;
+        return 1;
```

Comment:
It should be enough to just return 1 here, not bother to set any fields.

(I see that you may have followed what's done in py/objstr.c... so that code there could probably be simplified as well to just return 1.)